tiatoolbox/tools/tissuemask.py

```diff
@@ -0,0 +1,289 @@
+                e.g. regions of tissue vs background.
+        """
+        if not self.fitted:
+            raise Exception("Fit must be called before transform.")
+
+    def fit_transform(self, images: np.ndarray, **kwargs) -> np.ndarray:
+        """Perform :func:`fit` then :func:`transform`.
+
+        Sometimes it can be more optimal to perform both at the same
+        time for a single sample. In this case the base implementation
+        of :fun:`fit` followed by :func:`transform` can be overridden.
+
+        Args:
+            image (:class:`numpy.ndarray`): Image to create mask from.
+            kwargs (dict): Other key word arguments passed to fit.
+        """
+        self.fit(images, **kwargs)
+        return self.transform(images)
+
+
+class OtsuTissueMasker(TissueMasker):
+    """Tissue masker which uses Otsu's method to determine background.
+
+    Otsu's method
+
+    Examples:
+        >>> from tiatoolbox.tools.tissuemask import OtsuTissueMasker
+        >>> masker = OtsuTissueMasker()
+        >>> masker.fit(thumbnail)
+        >>> masks = masker.transform([thumbnail])
+
+        >>> from tiatoolbox.tools.tissuemask import OtsuTissueMasker
+        >>> masker = OtsuTissueMasker()
+        >>> masks = masker.fit_transform([thumbnail])
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.threshold = None
+
+    def fit(self, images: np.ndarray, masks=None) -> None:
+        """Find a binary threshold using Otsu's method.
+
+        Args:
+            images (:class:`numpy.ndarray`): List of images with a
+                length 4 shape (N, height, width, channels).
+            masks (:class:`numpy.ndarray`): Unused here, for API
+                consistency.
+        """
+        images_shape = np.shape(images)
+        if len(images_shape) != 4:
+            raise ValueError(
+                "Expected 4 dimensional input shape (N, height, width, 3)"
+                f" but recieved shape of {images_shape}."
+            )
+
+        # Convert RGB images to greyscale
+        grey_images = [x[..., 0] for x in images]
+        if images_shape[-1] == 3:
+            grey_images = np.zeros(images_shape[:-1], dtype=np.uint8)
+            for n, image in enumerate(images):
+                grey_images[n] = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
+
+        pixels = np.concatenate([np.array(grey).flatten() for grey in grey_images])
+
+        # Find Otsu's threshold for all pixels
+        self.threshold = threshold_otsu(pixels)
+
+        self.fitted = True
+
+    def transform(self, images: np.ndarray) -> np.ndarray:
+        """Create masks using the threshold found during :func:`fit`.
+
+
+        Args:
+            images (:class:`numpy.ndarray`): List of images with a
+                length 4 shape (N, height, width, channels).
+
+        Returns:
+            :class:`numpy.ndarray`: List of images with a length 4
+                shape (N, height, width, channels).
+        """
```

Comment:
Please add a new line at the end.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/5b0e0ecc1fcb9ce989d907483bef9a2b0b2c7aa5